src/CommonLib/Processors/WebClientServiceProcessor.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Win32.SafeHandles;
+using SharpHoundCommonLib.OutputTypes;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SharpHoundCommonLib.Processors
+{
+    public class WebClientServiceProcessor(ILogger log = null)
+    {
+        private readonly ILogger _log = log ?? Logging.LogProvider.CreateLogger("WebClientServiceProcessor");
+
+        // Define constants
+        public const uint MAXIMUM_ALLOWED = 0x02000000;
+        public const uint FILE_SHARE_READ = 0x00000001;
+        public const uint FILE_SHARE_WRITE = 0x00000002;
+        public const uint FILE_SHARE_DELETE = 0x00000004;
+        public const uint OPEN_EXISTING = 3;
+
+        // Error constants
+        public const uint ERROR_FILE_NOT_FOUND = 0x80070002;
+        public const uint ERROR_ACCESS_DENIED = 0x80070005;
+        public const uint ERROR_BAD_NETPATH = 0x80070035;
+        public const uint ERROR_NETNAME_DELETED = 0x80070040;
+        public const uint ERROR_NETWORK_ACCESS_DENIED = 0x80070041;
+        public const uint ERROR_LOGON_FAILURE = 0x8007052E;
+
+        [DllImport("kernel32.dll", SetLastError = true, CharSet = CharSet.Auto)]
+        public static extern SafeFileHandle CreateFile(
+            string lpFileName,
+            uint dwDesiredAccess,
+            uint dwShareMode,
+            IntPtr lpSecurityAttributes,
+            uint dwCreationDisposition,
+            uint dwFlagsAndAttributes,
+            IntPtr hTemplateFile);
+
+        public bool TestPathExists(string path)
+        {
+            using SafeFileHandle handle = CreateFile(
```

Comment:
Is there a benefit of this over using `File.Create`?

Reply:
Good question, @leechristensen can you clarify?

Reply:
I believe it was because File.Create always throws an exception for the pipe whereas we can get a success (and more granular error checking) with lower-level win32 APIs. I believe the error handling seemed simpler for Win32 APIs vs dealing with the various exceptions that .NET's File class may throw. But not tied to it either way.